pkg/controller/grafanadashboard/dashboard_pipeline.go

```diff
@@ -74,13 +74,8 @@ func (r *DashboardPipelineImpl) ProcessDashboard(knownHash string) ([]byte, erro
 	// Dashboards are never expected to come with an ID, it is
 	// always assigned by Grafana. If there is one, we ignore it
 	r.Board["id"] = nil
-
-	// This dashboard has previously been imported
-	// To make sure its updated we have to set the metadata
-	if r.Dashboard.Status.Phase == v1alpha1.PhaseReconciling {
-		r.Board["slug"] = r.Dashboard.Status.Slug
-		r.Board["uid"] = r.Dashboard.Status.UID
-	}
+	// Overwrite in case any user provided uid exists
+	r.Board["uid"] = nil
```

Comment:
Isn't this where we would specifically set the generated UID now?

Reply:
Yup, I do have a change locally that intends to overwrite this, but got sidetracked by figuring out some other change that I couldn't get working quite right, didn't want to push half-broken code